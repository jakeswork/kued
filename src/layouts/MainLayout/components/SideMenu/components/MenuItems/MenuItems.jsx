import React, { Fragment } from 'react';

import Button from '../../../../../../components/Button';

export default () => (
  <Fragment>
    <Button flat style={{ color: 'white' }} link="/feed">
      Warmane
    </Button>
    <Button flat style={{ color: 'white' }} link="/vote">
      Other Servers
    </Button>
    <Button flat style={{ color: 'white' }} link="/contact">
      Contact
    </Button>
  </Fragment>
);
