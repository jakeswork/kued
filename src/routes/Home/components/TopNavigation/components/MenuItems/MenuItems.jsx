import React, { Fragment } from 'react';

import Button from '../../../../../../components/Button';

export default () => (
  <Fragment>
    <Button flat style={{ color: 'white' }} link="/feed">
      Whos Playing
    </Button>
    <Button flat style={{ color: 'white' }} link="/vote">
      Compatibility
    </Button>
    <Button flat style={{ color: 'white' }} link="/contact">
      Contact
    </Button>
  </Fragment>
);
