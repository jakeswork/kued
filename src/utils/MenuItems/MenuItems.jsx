import React, { Fragment } from 'react';
import { node, string } from 'prop-types';

import Button from '../../components/Button';
import propTypes from './types';

const MenuItems = ({ classes }) => {
  const NavButton = ({ children, link }) => (
    <Button
      flat
      activeClassName={classes.active}
      link={link}
    >
      { children }
    </Button>
  );

  NavButton.propTypes = {
    children: node.isRequired,
    link: string.isRequired,
  };

  return (
    <Fragment>
      <NavButton link="/playing">
        Live Feed
      </NavButton>
      <NavButton link="/servers">
        Servers
      </NavButton>
      <NavButton link="/support">
        Support
      </NavButton>
    </Fragment>
  );
};

MenuItems.propTypes = propTypes;

export default MenuItems;
