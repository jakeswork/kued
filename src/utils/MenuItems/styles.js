export default theme => ({
  active: {
    color: theme.colorPrimary,
    cursor: 'default',
    pointerEvents: 'none',
  },
});
