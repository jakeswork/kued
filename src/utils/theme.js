export default {
  fontFamily: 'Karla,-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol',
  colorPrimary: '#fc3a3e',
  colorSecondary: '#17233a',
  colorGradientStart: '#fa7e32',
  colorGradientEnd: '#d1024f',
  colorRed: '#fb7979',
  colorGreen: '#1fd48e',
  colorBlue: '#00b4e6',
  colorAmber: '#fde9b4',
  textPrimary: '#2b090a',
  textSecondary: '#b1a0a0',
  textGrey: '#ccc',
  media: {
    mobile: '@media all and (max-width: 414px)',
  },
};
