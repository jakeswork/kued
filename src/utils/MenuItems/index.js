import withStyles from 'react-jss';

import MenuItems from './MenuItems';
import styles from './styles';

export default withStyles(styles)(MenuItems);
