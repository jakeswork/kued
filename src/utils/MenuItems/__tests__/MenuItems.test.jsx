import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'react-jss';
import { StaticRouter } from 'react-router-dom';

import MenuItems from '../MenuItems';
import theme from '../../theme';

const defaultProps = {
  classes: {},
};

describe('The MenuItems instance', () => {
  describe('when rendered', () => {
    it('should return the list of buttons', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <StaticRouter>
            <MenuItems {...defaultProps} />
          </StaticRouter>
        </ThemeProvider>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
