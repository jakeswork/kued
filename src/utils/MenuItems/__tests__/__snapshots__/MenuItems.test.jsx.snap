// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MenuItems instance when rendered should return the list of buttons 1`] = `
<ThemeProvider
  theme={
    Object {
      "colorAmber": "#fde9b4",
      "colorBlue": "#00b4e6",
      "colorGradientEnd": "#d1024f",
      "colorGradientStart": "#fa7e32",
      "colorGreen": "#1fd48e",
      "colorPrimary": "#fc3a3e",
      "colorRed": "#fb7979",
      "colorSecondary": "#17233a",
      "fontFamily": "Karla,-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
      "media": Object {
        "mobile": "@media all and (max-width: 414px)",
      },
      "textGrey": "#ccc",
      "textPrimary": "#2b090a",
      "textSecondary": "#b1a0a0",
    }
  }
>
  <StaticRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      staticContext={Object {}}
    >
      <MenuItems
        classes={Object {}}
      >
        <NavButton
          link="/playing"
        >
          <Jss(Button)
            analyticsLabel={null}
            danger={false}
            flat={true}
            icon={null}
            link="/playing"
            secondary={false}
            success={false}
          >
            <Button
              analyticsLabel={null}
              classes={
                Object {
                  "button": "Button-button-0-1-1 Button-button-0-1-6",
                  "buttonIcon": "Button-buttonIcon-0-1-5",
                  "flat": "Button-flat-0-1-4",
                  "link": "Button-link-0-1-3",
                  "secondary": "Button-secondary-0-1-2",
                }
              }
              danger={false}
              flat={true}
              icon={null}
              link="/playing"
              secondary={false}
              success={false}
              theme={
                Object {
                  "colorAmber": "#fde9b4",
                  "colorBlue": "#00b4e6",
                  "colorGradientEnd": "#d1024f",
                  "colorGradientStart": "#fa7e32",
                  "colorGreen": "#1fd48e",
                  "colorPrimary": "#fc3a3e",
                  "colorRed": "#fb7979",
                  "colorSecondary": "#17233a",
                  "fontFamily": "Karla,-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
                  "media": Object {
                    "mobile": "@media all and (max-width: 414px)",
                  },
                  "textGrey": "#ccc",
                  "textPrimary": "#2b090a",
                  "textSecondary": "#b1a0a0",
                }
              }
            >
              <NavLink
                className="Button-flat-0-1-4 Button-link-0-1-3"
                role="button"
                to="/playing"
              >
                <Route
                  path="\\\\/playing"
                >
                  <Link
                    aria-current={null}
                    className="Button-flat-0-1-4 Button-link-0-1-3"
                    role="button"
                    to="/playing"
                  >
                    <a
                      aria-current={null}
                      className="Button-flat-0-1-4 Button-link-0-1-3"
                      href="/playing"
                      onClick={[Function]}
                      role="button"
                    >
                      Live Feed
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </Button>
          </Jss(Button)>
        </NavButton>
        <NavButton
          link="/servers"
        >
          <Jss(Button)
            analyticsLabel={null}
            danger={false}
            flat={true}
            icon={null}
            link="/servers"
            secondary={false}
            success={false}
          >
            <Button
              analyticsLabel={null}
              classes={
                Object {
                  "button": "Button-button-0-1-1 Button-button-0-1-7",
                  "buttonIcon": "Button-buttonIcon-0-1-5",
                  "flat": "Button-flat-0-1-4",
                  "link": "Button-link-0-1-3",
                  "secondary": "Button-secondary-0-1-2",
                }
              }
              danger={false}
              flat={true}
              icon={null}
              link="/servers"
              secondary={false}
              success={false}
              theme={
                Object {
                  "colorAmber": "#fde9b4",
                  "colorBlue": "#00b4e6",
                  "colorGradientEnd": "#d1024f",
                  "colorGradientStart": "#fa7e32",
                  "colorGreen": "#1fd48e",
                  "colorPrimary": "#fc3a3e",
                  "colorRed": "#fb7979",
                  "colorSecondary": "#17233a",
                  "fontFamily": "Karla,-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
                  "media": Object {
                    "mobile": "@media all and (max-width: 414px)",
                  },
                  "textGrey": "#ccc",
                  "textPrimary": "#2b090a",
                  "textSecondary": "#b1a0a0",
                }
              }
            >
              <NavLink
                className="Button-flat-0-1-4 Button-link-0-1-3"
                role="button"
                to="/servers"
              >
                <Route
                  path="\\\\/servers"
                >
                  <Link
                    aria-current={null}
                    className="Button-flat-0-1-4 Button-link-0-1-3"
                    role="button"
                    to="/servers"
                  >
                    <a
                      aria-current={null}
                      className="Button-flat-0-1-4 Button-link-0-1-3"
                      href="/servers"
                      onClick={[Function]}
                      role="button"
                    >
                      Servers
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </Button>
          </Jss(Button)>
        </NavButton>
        <NavButton
          link="/support"
        >
          <Jss(Button)
            analyticsLabel={null}
            danger={false}
            flat={true}
            icon={null}
            link="/support"
            secondary={false}
            success={false}
          >
            <Button
              analyticsLabel={null}
              classes={
                Object {
                  "button": "Button-button-0-1-1 Button-button-0-1-8",
                  "buttonIcon": "Button-buttonIcon-0-1-5",
                  "flat": "Button-flat-0-1-4",
                  "link": "Button-link-0-1-3",
                  "secondary": "Button-secondary-0-1-2",
                }
              }
              danger={false}
              flat={true}
              icon={null}
              link="/support"
              secondary={false}
              success={false}
              theme={
                Object {
                  "colorAmber": "#fde9b4",
                  "colorBlue": "#00b4e6",
                  "colorGradientEnd": "#d1024f",
                  "colorGradientStart": "#fa7e32",
                  "colorGreen": "#1fd48e",
                  "colorPrimary": "#fc3a3e",
                  "colorRed": "#fb7979",
                  "colorSecondary": "#17233a",
                  "fontFamily": "Karla,-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
                  "media": Object {
                    "mobile": "@media all and (max-width: 414px)",
                  },
                  "textGrey": "#ccc",
                  "textPrimary": "#2b090a",
                  "textSecondary": "#b1a0a0",
                }
              }
            >
              <NavLink
                className="Button-flat-0-1-4 Button-link-0-1-3"
                role="button"
                to="/support"
              >
                <Route
                  path="\\\\/support"
                >
                  <Link
                    aria-current={null}
                    className="Button-flat-0-1-4 Button-link-0-1-3"
                    role="button"
                    to="/support"
                  >
                    <a
                      aria-current={null}
                      className="Button-flat-0-1-4 Button-link-0-1-3"
                      href="/support"
                      onClick={[Function]}
                      role="button"
                    >
                      Support
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </Button>
          </Jss(Button)>
        </NavButton>
      </MenuItems>
    </Router>
  </StaticRouter>
</ThemeProvider>
`;
